### PR TITLE
Feat(prql): Add support for TAKE 

### PR DESCRIPTION
**Pull request by fool1280**
[PRQL take transform doc](https://prql-lang.org/book/reference/stdlib/transforms/take.html)

### CODE REVIEW

File: sqlglot/dialects/prql.py [hunk 1]
```diff
@@ -78,6 +79,22 @@ def _parse_selection(self, query: exp.Query, append: bool = True) -> exp.Query:
 
             return query.select(*selects, append=append, copy=False)
 
+        def _parse_take(self, query: exp.Query) -> exp.Query:
+            if self._match(TokenType.NUMBER):
+                first = self._prev.text  # TAKE 4..10 -> self._prev.text = "4."
```
Comment (georgesittas) [inline sqlglot/dialects/prql.py:84]: This doesn't look right. We probably need to tokenize `1..10` differently. One way is to produce `[1, .., 10]`, i.e. create a `DDOT` token that represents `".."` and then add a method that parses `NUMBER DDOT NUMBER` into a `Slice` or something.

File: sqlglot/dialects/prql.py [hunk 2]
```diff
@@ -78,6 +79,22 @@ def _parse_selection(self, query: exp.Query, append: bool = True) -> exp.Query:
...
+            else:
+                self.raise_error("Expecting a number")
+
+            if self._match_pair(TokenType.DOT, TokenType.NUMBER):  # TAKE <start>..<end>
+                first = int(first[:-1])
+                second = int(self._prev.text)
+                if second >= first:  # valid range
+                    return query.limit(second - first + 1).offset(first - 1)
+                else:
+                    return query.limit(0)
```
Comment (georgesittas) [inline sqlglot/dialects/prql.py:94]: Why is this doing `limit(0)`?

Comment (fool1280) [reply sqlglot/dialects/prql.py:94]: From the [PRQL Playground](https://prql-lang.org/playground/), the `from x take 1..0` seems to be a valid query and produce `select * from x limit 0`, so I want to handle that here.

File: sqlglot/dialects/prql.py [hunk 3]
```diff
@@ -78,6 +79,11 @@ def _parse_selection(self, query: exp.Query, append: bool = True) -> exp.Query:
 
             return query.select(*selects, append=append, copy=False)
 
+        def _parse_take(self, query: exp.Query) -> exp.Query:
+            if self._match(TokenType.NUMBER):
+                val = self._prev.text  # TAKE <NUMBER>
+            return query.limit(int(val))  # TODO: TAKE for ranges a..b
```
Comment (georgesittas) [inline sqlglot/dialects/prql.py:85]: Let's simplify this to:

```suggestion
            return query.limit(self._parse_number())  # TODO: TAKE for ranges a..b
```

**Review by georgesittas — APPROVED** [review]
One last comment and should be good to go

File: sqlglot/dialects/prql.py (hunk 3 above)
Comment (fool1280) [reply sqlglot/dialects/prql.py]: thanks George!

Comment (georgesittas) [reply sqlglot/dialects/prql.py]: Sure thing! I guess we need to update the type hint so it's optional as well.

**Pull request merged**